obstore/python/obstore/fsspec.py

```diff
@@ -296,3 +468,66 @@ def read(self, length: int = -1) -> Any:
             data = self.fs.cat_file(self.path, self.loc, self.loc + length)
             self.loc += length
         return data
+
+
+def register(protocol: str | list[str], asynchronous: bool = False) -> None:
```

Comment:
```suggestion
def register(protocol: str | list[str], *, asynchronous: bool = False) -> None:
```
I'm a big proponent of having a small number of positional parameters. This also allows us to make `asynchronous` a positional argument in the future without it being breaking